pkg/wasm/module_getter.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	adminv3 "github.com/envoyproxy/go-control-plane/envoy/admin/v3"
+	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	wasm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/wasm/v3"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"istio.io/istio/pkg/util/configdump"
+	"istio.io/istio/pkg/util/sets"
+)
+
+type ModuleGetter interface {
+	ListUsingModules() (sets.String, error)
+}
+
+type LocalModuleGetter struct {
+	AdminPort     int32
+	LocalHostAddr string
+}
+
+func NewLocalModuleGetter(adminPort int32, localhostAddr string) ModuleGetter {
+	return &LocalModuleGetter{
+		AdminPort:     adminPort,
+		LocalHostAddr: localhostAddr,
+	}
+}
+
+func (g *LocalModuleGetter) ListUsingModules() (sets.String, error) {
```

Comment:
I am concerned about the perf

Reply:
+1, I am pretty strongly against using config_dump in anything other than debugging context.

*if* we did, there is a way to filter it though: https://www.envoyproxy.io/docs/envoy/latest/operations/admin#get--config_dump?mask=

Reply:
I agree with that `config_dump is not an elegant way`, and we need more experiments before introducing `TTL`

Speaking of performance issues, interval should not be very short in a real world, but as your said with `mask` will be better.